lib/HALTarget/src/Device.cpp

```diff
@@ -0,0 +1,107 @@
+/* MIT License
+ *
+ * Copyright (c) 2023 Andreas Merkle <web@blue-andi.de>
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+/*******************************************************************************
+    DESCRIPTION
+*******************************************************************************/
+/**
+ * @brief  Device realization
+ * @author Gabryel Reyes <gabryelrdiaz@gmail.com>
+ */
+
+/******************************************************************************
+ * Includes
+ *****************************************************************************/
+#include "Device.h"
+#include "GPIO.h"
+
+/******************************************************************************
+ * Compiler Switches
+ *****************************************************************************/
+
+/******************************************************************************
+ * Macros
+ *****************************************************************************/
+
+/******************************************************************************
+ * Types and classes
+ *****************************************************************************/
+
+/******************************************************************************
+ * Prototypes
+ *****************************************************************************/
+
+/******************************************************************************
+ * Local Variables
+ *****************************************************************************/
+
+/******************************************************************************
+ * Public Methods
+ *****************************************************************************/
+
+bool Device::init()
+{
+    reset();
+    return m_usbHost.init();

```

Comment:
This will fail because the reset line is being held high by reset() until process() gets called and the timer is timed-out. Would a blocking reset of RESET_TIME_MS be too much during initialization?

```cpp
GpioPins::resetDevicePin.write(HIGH);
delay(RESET_TIME_MS)
GpioPins::resetDevicePin.write(LOW);
```

Reply:
The return value is ok, I was just asking because original the return value was not verified. :-)

Reply:
Sorry, your answer does not completely resolve my question. 
During init(), the reset() method pulls the reset line of the device HIGH, and keeps it that way until the timer is done.
If the reset line is HIGH, m_usbHost.init() will fail. The line must be pulled LOW.
Would the suggested code above be a viable solution?

Reply:
No, because as long as the timer is running, it will fail.

Reply:
I checked again and this is not necessary. I was wrong. The init of the USB host does not require the USB client to be enabled.